### PR TITLE
Skip vf configuration when not found in VF group

### PR DESCRIPTION
**Pull request by mlguerrero12**

### CODE REVIEW

File: pkg/utils/utils.go [hunk 1]
```diff
@@ -476,6 +478,11 @@ func configSriovDevice(iface *sriovnetworkv1.Interface, ifaceStatus *sriovnetwor
 				}
 			}
 
+			// Do not configure VF which is not part of any VF Group.
+			if !found {
+				continue
+			}
```
Comment (bn222) [inline pkg/utils/utils.go:484]: This is essentially filtering the list of addresses to process. Now, the body of the loop does much too much imo. Let's take the opportunity to refactor this whole function to have a more proper way to filter the list

Comment (mlguerrero12) [reply pkg/utils/utils.go:479]: done, please have a look again

**Comment by mlguerrero12** [comment]
created a test in https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/541 to show the issue. I added here as well. It should pass now.

**Comment by mlguerrero12** [comment]
OCP job passed (with new added test). There is a problem setting up the k8s job.

File: pkg/utils/utils.go [hunk 2]
```diff
@@ -458,24 +458,26 @@ func configSriovDevice(iface *sriovnetworkv1.Interface, ifaceStatus *sriovnetwor
...
-			for i, group = range iface.VfGroups {
-				if err != nil {
-					log.Log.Error(err, "configSriovDevice(): unable to get VF id", "device", iface.PciAddress)
-				}
-				if sriovnetworkv1.IndexInRange(vfID, group.VfRange) {
-					isRdma = group.IsRdma
-					if sriovnetworkv1.StringInArray(group.DeviceType, DpdkDrivers) {
-						dpdkDriver = group.DeviceType
-					}
+			if err != nil {
```
Comment (adrianchiris) [inline pkg/utils/utils.go:464]: thx for moving this check !

**Review by adrianchiris — APPROVED** [review]

File: test/conformance/tests/test_sriov_operator.go [hunk 3]
```diff
@@ -1071,6 +1071,83 @@ var _ = Describe("[sriov] operator", func() {
 					}))
 				})
 
+				It("Should configure the mtu only for vfs which are part of the partition", func() {
```
Comment (zeeke) [inline test/conformance/tests/test_sriov_operator.go:1074]: Thanks for moving the test into this PR!

**Review by zeeke — APPROVED** [review]
LGTM

Thanks @mlguerrero12

**Pull request merged**